signale.d.ts

```diff
@@ -0,0 +1,148 @@
+        /** Sets the configuration of an instance overriding any existing global or local configuration. */
+        config?: SignaleConfig;
+        /**
+         * Name of the scope.
+         */
+        scope?: string;
+        /**
+         * Holds the configuration of the custom and default loggers.
+         */
+        types?: Partial<Record<TTypes, CommandType>>;
+        interactive?: boolean;
+        timers?: Map<string, Date>;
+        /**
+         * Destination to which the data is written, can be any valid
+         * [Writable stream](https://nodejs.org/api/stream.html#stream_writable_streams).
+         */
+        stream?: NodeJS.WriteStream;
+    }
+
+    interface SignaleConstructor {
+        new <TTypes extends string = DefaultMethods>(
+            options?: SignaleOptions<TTypes>
+        ): Signale<TTypes>;
+    }
+
+    interface SignaleBase<TTypes extends string = DefaultMethods> {
+        /**
+         * Sets the configuration of an instance overriding any existing global or local configuration.
+         *
+         * @param configObj Can hold any of the documented options.
+         */
+        config(configObj: SignaleConfig): Signale<TTypes>;
+        /**
+         * Defines the scope name of the logger.
+         *
+         * @param name Can be one or more comma delimited strings.
+         */
+        scope(...name: string[]): Signale<TTypes>;
+        /** Clears the scope name of the logger. */
+        unscope(): void;
+        /**
+         * Sets a timers and accepts an optional label. If none provided the timer will receive a unique label automatically.
+         *
+         * @param label Label corresponding to the timer. Each timer must have its own unique label.
+         * @returns a string corresponding to the timer label.
+         */
+        time(label?: string): string;
+        /**
+         * Deactivates the timer to which the given label corresponds. If no label
+         * is provided the most recent timer, that was created without providing a
+         * label, will be deactivated.
+         *
+         * @param label Label corresponding to the timer, each timer has its own unique label.
+         * @param span Total running time.
+         */
+        timeEnd(
+            label?: string,
+            span?: number
+        ): { label: string; span?: number };
+    }
+
+    type LoggerFunc = (message?: any, ...optionalArgs: any[]) => void;
+    type Signale<TTypes extends string = DefaultMethods> = SignaleBase<TTypes> &
+        Record<TTypes, LoggerFunc> &
+        Record<DefaultMethods, LoggerFunc>;
+}
+
+declare const Signale: Signale.Signale<Signale.DefaultMethods> & {
+    Signale: Signale.SignaleConstructor;
+    SignaleConfig: Signale.SignaleConfig;
+    SignaleOptions: Signale.SignaleOptions;
+    DefaultMethods: Signale.DefaultMethods;
+};
+
+declare function signale<T>(opt?: Signale.SignaleOptions): Signale.Signale<Signale.DefaultMethods> & {
```

Comment:
Is this `declare function` necessary, btw? Shouldn't the above `declare const` solve the commonjs default export issue?

Reply:
I was trying to solve the `common.js default export issue` by this. I was following `fastify` definition file. If you say I could remove that.